app/main.py

```diff
@@ -1,3 +1,47 @@
+from functools import total_ordering
+from typing import Union
+
+
+@total_ordering
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, "Distance"]) -> "Distance":
```

Comment:
```suggestion
    def __add__(self, other: Union[int, float, Distance]) -> Distance:
```

Reply:
use from __future__ import annotations

Reply:
If I remove the quotes, it gives the error "Unresolved reference 'Distance'"